core/src/main/scala/org/apache/spark/deploy/master/Master.scala

```diff
@@ -550,6 +502,55 @@ private[deploy] class Master(
       } else {
         context.reply(0)
       }
+
+    case ExecutorStateChanged(appId, execId, state, message, exitStatus) =>
```

Comment:
This change moves the handling of `ExecutorStateChanged` from `receive` to `receiveAndReply` and `context.reply(true)` is the only difference.